### DISCUSSION TITLE
You have exceeded a secondary rate limit and have been temporarily blocked from content creation. Please retry your request again later.

### DISCUSSION THREAD

**Discussion by AutomationUIUser** [community/community#32120] [Apps, API and Webhooks]
When trying to create multiple issues from API, creation gets blocked after some operations. Getting below error when trying to process the API.

**"message":"You have exceeded a secondary rate limit and have been temporarily blocked from content creation. Please retry your request again later."**

At this time API header response, I am getting is below:

[NameValuePair(key=Server, value=GitHub.com), NameValuePair(key=Date, value=Mon, 05 Sep 2022 07:53:41 GMT), NameValuePair(key=Content-Type, value=application/json; charset=utf-8), NameValuePair(key=Transfer-Encoding, value=chunked), NameValuePair(key=X-OAuth-Scopes, value=admin:enterprise, admin:gpg_key, admin:org, admin:org_hook, admin:public_key, admin:repo_hook, delete:packages, delete_repo, repo, user, workflow, write:discussion, write:packages), NameValuePair(key=X-Accepted-OAuth-Scopes, value=), NameValuePair(key=X-GitHub-Media-Type, value=github.v3; format=json), NameValuePair(key=X-RateLimit-Limit, value=5000), NameValuePair(key=X-RateLimit-Remaining, value=4946), NameValuePair(key=X-RateLimit-Reset, value=1662367828), NameValuePair(key=X-RateLimit-Used, value=54), NameValuePair(key=X-RateLimit-Resource, value=core), NameValuePair(key=Access-Control-Expose-Headers, value=ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset), NameValuePair(key=Access-Control-Allow-Origin, value=*), NameValuePair(key=Strict-Transport-Security, value=max-age=31536000; includeSubdomains; preload), NameValuePair(key=X-Frame-Options, value=deny), NameValuePair(key=X-Content-Type-Options, value=nosniff), NameValuePair(key=X-XSS-Protection, value=0), NameValuePair(key=Referrer-Policy, value=origin-when-cross-origin, strict-origin-when-cross-origin), NameValuePair(key=Content-Security-Policy, value=default-src 'none'), NameValuePair(key=Vary, value=Accept-Encoding, Accept, X-Requested-With), NameValuePair(key=X-GitHub-Request-Id, value=77D2:7B0D:1C580A:203694:6315AB04)]

Here, we can see that **X-RateLimit-Remaining's value=4946**. So, the rate limit is remaining but still create operation is blocked.

**What is the reason for this blocking? And is there a way from the header from which I can identify the failure cause? How I can differentiate this blocking from the normal rate limit exceeding blocking?**

**Comment by byrneh** [comment]
@AutomationUIUser there are dfferent types of rate limits, see [#rate-limiting](https://docs.github.com/en/rest/overview/resources-in-the-rest-api#rate-limiting). 
Additionally there is some general guidance around secondary rate limits here [#dealing-with-secondary-rate-limits](https://docs.github.com/en/rest/guides/best-practices-for-integrators#dealing-with-secondary-rate-limits).  The scondary rate limit criteria is not based solely on volume of request (X-RateLimit-Remaining, but on the complexisty and type of requests and data processed.

**Reply by darrell-hart** [reply]
I am seeing this too.  Unfortunately the guidance on [#dealing-with-secondary-rate-limits](https://docs.github.com/en/rest/guides/best-practices-for-integrators#dealing-with-secondary-rate-limits) is inaccurate as it states you will get a `Retry-After` value that you can use.   This isn't the case though as you can see from the response above.  Without this value how do you know how long to wait?

**Reply by g0d33p3rsec** [reply]
> I am seeing this too. Unfortunately the guidance on [#dealing-with-secondary-rate-limits](https://docs.github.com/en/rest/guides/best-practices-for-integrators#dealing-with-secondary-rate-limits) is inaccurate as it states you will get a `Retry-After` value that you can use. This isn't the case though as you can see from the response above. Without this value how do you know how long to wait?

there doesn't seem to be any practical guidance, just wait and hope.... typical MSFT antifeature
from https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28
"These secondary rate limits are subject to change without notice. You may also encounter a secondary rate limit for undisclosed reasons."

example"
```
ERROR: 403 {"message":"You have exceeded a secondary rate limit and have been temporarily blocked from content creation. Please retry your request again later. If you reach out to GitHub Support for help, please include the request ID 8CFE:8AB22:427A525:8495302:67782E88 and timestamp 2025-01-03 18:38:00 UTC.","documentation_url":"https://docs.github.com/rest/overview/rate-limits-for-the-rest-api#about-secondary-rate-limits","status":"403"}
```

```
 https://api.github.com/rate_limit
{
  "resources": {
    "core": {
      "limit": 5000,
      "used": 0,
      "remaining": 5000,
      "reset": 1735933569
    },
    "search": {
      "limit": 30,
      "used": 0,
      "remaining": 30,
      "reset": 1735930029
    },
    "graphql": {
      "limit": 5000,
      "used": 17,
      "remaining": 4983,
      "reset": 1735930338
    },
    "integration_manifest": {
      "limit": 5000,
      "used": 0,
      "remaining": 5000,
      "reset": 1735933569
    },
    "source_import": {
      "limit": 100,
      "used": 0,
      "remaining": 100,
      "reset": 1735930029
    },
    "code_scanning_upload": {
      "limit": 1000,
      "used": 0,
      "remaining": 1000,
      "reset": 1735933569
    },
    "code_scanning_autofix": {
      "limit": 10,
      "used": 0,
      "remaining": 10,
      "reset": 1735930029
    },
    "actions_runner_registration": {
      "limit": 10000,
      "used": 0,
      "remaining": 10000,
      "reset": 1735933569
    },
    "scim": {
      "limit": 15000,
      "used": 0,
      "remaining": 15000,
      "reset": 1735933569
    },
    "dependency_snapshots": {
      "limit": 100,
      "used": 0,
      "remaining": 100,
      "reset": 1735930029
    },
    "audit_log": {
      "limit": 1750,
      "used": 0,
      "remaining": 1750,
      "reset": 1735933569
    },
    "audit_log_streaming": {
      "limit": 15,
      "used": 0,
      "remaining": 15,
      "reset": 1735933569
    },
    "code_search": {
      "limit": 10,
      "used": 0,
      "remaining": 10,
      "reset": 1735930029
    }
  },
  "rate": {
    "limit": 5000,
    "used": 0,
    "remaining": 5000,
    "reset": 1735933569
  }
}
```

**Comment by johnbent** [comment]
This is super-annoying. I run a volunteer non-profit webpage for the preservation of an endangered language. We have a volunteer group who consults about words which have not yet been documented. We used to use debugle for these discussions but debugle closed. No problem; github is awesome and free. But I wrote a script to automatically create issues in github from the debugle issues and I keep hitting the secondary rate limit. It shouldn't be a problem because I only need to create about 100 issues with 5 comments each. But I keep hitting this limit. It wouldn't be a problem if I could query the limit and then sleep accordingly. But when I try to query, all my rates look OK!!! Argh. :(

~/tekinged.com/scripts/debugle> curl   -H "Accept: application/vnd.github+json"   -H "Authorization: Bearer $GH_OATH"   https://api.github.com/rate_limit | grep remaining 
      "remaining": 4715,
      "remaining": 30,
      "remaining": 5000,
      "remaining": 5000,
      "remaining": 100,
      "remaining": 1000,
      "remaining": 10000,
      "remaining": 15000,
      "remaining": 100,
    "remaining": 4715,
